src/Bicep.Core/TypeSystem/TypeManager.cs

```diff
@@ -18,15 +18,18 @@ public class TypeManager : ITypeManager
     {
         private readonly IReadOnlyDictionary<SyntaxBase,Symbol> bindings;
 
+        private readonly IReadOnlyDictionary<SyntaxBase, DeclaredSymbol[]> cyclesBySytax;
+
         // stores results of type checks
         private readonly ConcurrentDictionary<SyntaxBase, TypeSymbol> typeCheckCache = new ConcurrentDictionary<SyntaxBase, TypeSymbol>();
 
-        public TypeManager(IReadOnlyDictionary<SyntaxBase, Symbol> bindings)
+        public TypeManager(IReadOnlyDictionary<SyntaxBase, Symbol> bindings, IReadOnlyDictionary<SyntaxBase, DeclaredSymbol[]> cyclesBySytax)
```

Comment:
>DeclaredSymbol[] [](start = 109, length = 16)

This is mutable, but I'm not sure what to do about it, though :(

Reply:
Make it immutable! For some reason my thought process was that because you can't append to an array, it is immutable. But obviously you can overwrite items...